### PR TITLE
[STORY-642] new api endpoint to handle password reset

### PR DESCRIPTION
**Pull request by sc-zenokerr**
- [ ] Add a changelog entry in the section "To Be Released" of CHANGELOG.md

Database user password update uses new API endpoint for generating password.

This is dependent on the merging / release of the equivalent branch in go-scalingo library.

[Story 642](https://www.notion.so/scalingooriginal/New-API-endpoint-to-handle-password-reset-662bd5bb9ec045308e5348f499055e65)

### CODE REVIEW

File: db/users/update_password.go [hunk 1]
```diff
@@ -60,25 +59,31 @@ func UpdateUser(ctx context.Context, app, addonUUID, username string) error {
...
+		userUpdateParam := scalingo.DatabaseUpdateUserParam{
+			DatabaseID:           addonUUID,
+			Password:             password,
+			PasswordConfirmation: confirmedPassword,
+		}
+		databaseUser, err = c.DatabaseUpdateUser(ctx, app, addonUUID, username, userUpdateParam)
+		if err != nil {
+			return errors.Wrap(ctx, err, "update password of the given database user")
+		}
+	} else {
```
Comment (EtienneM) [inline db/users/update_password.go:77]: suggestion: I like returning the function as soon as possible. In this case, you could return here and the `else` is useless. There is less indentation which makes the code easier to read IMO
```suggestion
		}
		
		fmt.Printf("User \"%s\" password updated.\n", databaseUser.Name)
		return nil
	}
```

**Review by EtienneM — APPROVED** [review]
Just a small suggestion but LGTM :+1: 

Did you ask product about the CLI backward compatibility?

File: db/users/update_password.go (hunk 1 above)
Comment (sc-zenokerr) [reply db/users/update_password.go]: Thanks. I actually misread the original flow. Your suggestion is absolutely correct. Thanks

Comment (sc-zenokerr) [reply db/users/update_password.go]: > Did you ask product about the CLI backward compatibility?

Yes. My question was answered by benjamin in the comments in the [STORY-642] in notion. 

It should be (remain) possible to set a user-supplied password using the CLI.

**Review by EtienneM — APPROVED** [review]

**Pull request merged**